qt/python/mantidqt/utils/asynchronous.py

```diff
@@ -100,6 +59,77 @@ def elapsed(start):
 
         self.finished_cb()
 
+    def abort(self):
+        """Cancel an asynchronous execution"""
+        # Implementation is based on
+        # https://stackoverflow.com/questions/5019436/python-how-to-terminate-a-blocking-thread
+        ctypes.pythonapi.PyThreadState_SetAsyncExc(ctypes.c_long(self.ident),
+                                                   ctypes.py_object(KeyboardInterrupt))
+        time.sleep(0.1)
+
+
+class BlockingAsyncTask(AsyncTask):
+    def __init__(self, target, args=(), kwargs=None, success_cb=None, error_cb=None, blocking_cb=None,
+                 period_secs=0.05):
+        """Run the target in a separate thread and block the calling thread
+        until execution is complete.
+
+        :param target: A Python callable object
+        :param args: Arguments to pass to the callable
+        :param kwargs: Keyword arguments to pass to the callable
+        :param blocking_cb: An optional callback to process while waiting for the task
+        to finish
+        :param period_secs: Sleep for this many seconds at the start of each loop that checks
+        the task is still alive. This will be the minimum time between calls to blocking_cb.
+        :returns: An AsyncTaskResult object
+        """
+        super(BlockingAsyncTask, self).__init__(target)
+
+        self.period_secs = period_secs
+
+        def create_callback(cb):
+            return cb if cb is not None else lambda: None
+
+        self.blocking_cb = create_callback(blocking_cb)
+        self.success_cb = create_callback(success_cb)
+        self.error_cb = create_callback(error_cb)
+
+        class Receiver(object):
+            output, exc_value = None, None
+
+            def on_success(self, result):
+                self.output = result.output
+                if success_cb is not None:
+                    success_cb(result)
+
+            def on_error(self, result):
+                self.exc_value = result.exc_value
+                if error_cb is not None:
+                    error_cb(result)
+
+        self.recv = Receiver()
+        self.task = AsyncTask(target, args, kwargs, success_cb=self.recv.on_success, error_cb=self.recv.on_error)
+
+    def begin(self):
```

Comment:
The idea is that it would not overload super.start hence using a different name, it fundamentally does a different operation, even though the words may mean similar things in this context.

Reply:
Wouldn't `.start` then run the BlockingAsync task just Async without blocking? If this is the case then it should definitely overloaded to prevent that behaviour, and callers should use `AsyncTask` if they want async.